solutions/ac-MultipleOf2019.mdx

```diff
@@ -2,11 +2,32 @@
 id: ac-MultipleOf2019
 source: AtCoder Beginner Contest
 title: Multiple of 2019
-author: Maggie Liu, Brad Ma, Mohammad Nour Massri
+author: Maggie Liu, Brad Ma, Mohammad Nour Massri, David Zhang
 ---
 
 [Official Analysis](https://img.atcoder.jp/abc164/editorial.pdf#page=10)
 
+## Explanation
+First of all, in order to store sections of the string, we can make a suffix array with the condition that at index $$i$$ in suffix array $${S}$$
+
+$${S_i = s_n + 10*s_{n-1} +100*s_{n-2} + ... + {10^{n-i}} * s_{i}}$$
+
+For example, in the string "1234", the suffix array would be
+
+[1234, 234, 34, 4]
+
+This is useful because we can query any substring in $\mathcal{O}(1)$ time. So if $${S_0} = 1234 $$ and $${S_3=4}$$, this means $${S_0}-{S_3} = 123 * {10^1}$$ or $$1230$$. 
```

Comment:
what's a "query"?

Reply:
range query, so S_i - S_j will allow us to get a number from a certain substring

Reply:
> don't explain it to me, explain it to the users lol

Reply:
ofc